### DISCUSSION TITLE
Composite .svg and .jpg images

### DISCUSSION THREAD

**Discussion by lusy777** [ImageMagick/ImageMagick#1929] [General]
Hello!
I'm trying to merge two different image formats. The bottom layer must be an jpg (or png) image, the top layer must be svg image (In this image is just a text without background and this should be overlaid on the first image). I tried to apply the command:
`composite 1.svg 2.jpg 0.jpg`
but svg image is inserted as a white background on the second image and overlaps it. Please help!

**Comment by snibgo** [comment]
ImageMagick is a raster processor. If you read a SVG (vector) image, IM will rasterize it.

What you seem to want is a SVG file that contains a vector image. SVG files are text, so they are easy to create. If you don't know SVG syntax, the easy way to create them is with Inkscape. Yoy can create text or other vector elements, and embed whatever raster images you want.

**Reply by lusy777** [reply]
Thanks for reply! I need to programmatically combine two images..
How do I do better for now? to transform the jpg image to svg and combine the both svg images or .. ?

**Comment by snibgo** [comment]
Make a text file with a base64 encoding of your raster image. For example, using the built-in "rose:" image:

`magick rose: inline:png:x.txt`

The file x.txt will start with something like this:
`data:image/png;base64,iVBORw0KGg`

Add a header and trailer, so the result starts with:
`<image
     x="0"
     y="0"
     width="70"
     height="46"
     id="image1"
     xlink:href="data:image/png;base64,iVBORw0KGg`
... and a new line at the bottom is:
`==" />`

The x and y attributes are the position you want your raster image to appear within the SVG. The width and height attributes are the dimensions of the raster image.

So this is a SVG "image" element. You can insert it anywhere you want in your SVG file, most likely near the top.
